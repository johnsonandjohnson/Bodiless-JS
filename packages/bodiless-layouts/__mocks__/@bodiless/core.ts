const editContext = {
  activate: jest.fn(),
  isEdit: true,
};

export const useEditContext = jest.fn(() => editContext);

const activateOnEffect = {
  setId: jest.fn(),
};
export const useActivateOnEffect = () => activateOnEffect;

export const useNode = jest.fn();

const contextMenuFormInner = jest.fn();
export const contextMenuForm = jest.fn(() => contextMenuFormInner);
