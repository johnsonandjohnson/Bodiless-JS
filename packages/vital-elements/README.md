# `@bodiless/vital-elements`

For documentation on the Vital Elements package, please see:

- [Vital Elements | BodilessJS Docs](https://johnsonandjohnson.github.io/Bodiless-JS/#/VitalDesignSystem/Components/VitalElements/)
- [`doc/` Directory](./doc)
