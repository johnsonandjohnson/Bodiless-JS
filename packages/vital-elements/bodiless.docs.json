{
  "VitalDesignSystem": {
    "README.md": "./doc/VitalDesignSystem.md",
    "GettingStarted": {
      "README.md": "./doc/GettingStarted/README.md",
      "CoreConcepts.md": "./doc/GettingStarted/CoreConcepts.md",
      "ComponentsAndTokens.md": "./doc/GettingStarted/ComponentsAndTokens.md",
      "CompositionFromWithout.md": "./doc/GettingStarted/ComposingFromOutside.md",
      "ReachingInside.md": "./doc/GettingStarted/ReachingInside.md",
      "AnatomyOfA_Token.md": "./doc/GettingStarted/AnatomyOfAToken.md",
      "FunctionalClasses.md": "./doc/GettingStarted/FunctionalClasses.md",
      "WorkingWithContent.md": "./doc/GettingStarted/WorkingWithContent.md",
      "CleanComponents.md": "./doc/GettingStarted/CleanComponents.md",
      "YourFirstToken.md": "./doc/GettingStarted/YourFirstToken.md",
      "WhyVital.md": "./doc/GettingStarted/WhyVital.md"
    },
    "Components": {
      "README.md": "./doc/Components.md",
      "VitalElements": {
        "README.md": "./doc/VitalElements.md",
        "ComponentTemplate.md": "./doc/ComponentTemplate.md",
        "ExtendingAndComposingTokens.md": "./doc/ExtendingAndComposing.md",
        "SiteTypography.md": "./doc/SiteTypography.md",
        "Shadow.md": "./doc/Shadow.md",
        "StaticReplacement.md": "./doc/StaticReplacement.md",
        "Tokens": {
          "README.md": "./doc/Tokens.md",
          "TokenDomains.md": "./doc/TokenDomains.md"
        }
      },
      "assets": {
        "PXtoREMTemplate.xlsx": "./doc/assets/PXtoREMTemplate.xlsx"
      }
    }
  }
}
