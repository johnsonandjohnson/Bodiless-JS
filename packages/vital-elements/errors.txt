Errors for Brand Tokens/White Label/Semantic/Color/Background/Section/Light Theme/Secondary
 . Attempt to resolve alias to undefined
Errors for Brand Tokens/White Label/Semantic/Color/Background/Section/Light Theme/Tertiary
 . Attempt to resolve alias to undefined
Errors for Brand Tokens/White Label/Components/Checkbox/Background/Dark Theme/Hover
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Border/Dark Theme/Disabled
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Border/Dark Theme/Checked
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Border/Dark Theme/Hover
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Border/Dark Theme/Idle
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Background/Light Theme/Idle
 . Component color alias "(Color/--NO VALUE--" is not semantic
Errors for Brand Tokens/White Label/Components/Checkbox/Background/Dark Theme/Disabled
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Background/Dark Theme/Idle
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Background/Dark Theme/Checked
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Form Field/Text/Caption
 . Component color alias "(Typography/Body/Small" is not semantic
Errors for Brand Tokens/White Label/Components/Form Field/Border/Error
 . Component color alias "(Color/Signal/Error/Error" is not semantic
Errors for Brand Tokens/White Label/Components/Form Field/Text/Error
 . Component color alias "(Color/Signal/Error/Error" is not semantic
Errors for Brand Tokens/White Label/Components/Form Field/Border/Success
 . Component color alias "(Color/Signal/Success/Success" is not semantic
Errors for Brand Tokens/White Label/Components/Form Field/Text/Input
 . Component color alias "(Typography/Body/Regular" is not semantic
Errors for Brand Tokens/White Label/Components/Form Field/Text/Label
 . Component color alias "(Typography/Eyebrow/Eyebrow" is not semantic
Errors for Brand Tokens/White Label/Components/Navigation/Link
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Navigation/Primary/Primary Link/Border/Hover
 . Component color alias "(Semantic/Color/Interactive/Default" is not semantic
Errors for Brand Tokens/White Label/Components/Navigation/Primary/Primary Link/Border/Active
 . Component color alias "(Semantic/Color/Interactive/Hover" is not semantic
Errors for Brand Tokens/White Label/Components/Category Card/Padding/Bottom
 . Spacing value "0" is not an alias
Errors for Brand Tokens/White Label/Components/Category Card/Padding/Top
 . Spacing value "0" is not an alias
Errors for Brand Tokens/White Label/Components/Category Card/Background/Light Theme/Background
 . Component color alias "(Color/--NO VALUE--" is not semantic
Errors for Brand Tokens/White Label/Components/Category Card/Background/Dark Theme/Background
 . Component color alias "(Color/--NO VALUE--" is not semantic
Errors for Brand Tokens/White Label/Components/Accordion/Dropdown/Border/Light Theme/Border
 . Component color alias "(Color/--NO VALUE--" is not semantic
Errors for Brand Tokens/White Label/Components/Breadcrumb/Background/Light Theme/Color
 . Component color alias "(Semantic/Color/Background/Page/Primary" is not semantic
Errors for Brand Tokens/White Label/Components/Breadcrumb/Background/Dark Theme/Color
 . Component color alias "(Semantic/Color/Background/Page/Primary" is not semantic
Errors for Brand Tokens/White Label/Components/Article Card/Border/Dark Theme/Background
 . Component color alias "(Color/--NO VALUE--" is not semantic
Errors for Brand Tokens/White Label/Components/Article Card/Border/Light Theme/Background
 . Component color alias "(Color/--NO VALUE--" is not semantic
