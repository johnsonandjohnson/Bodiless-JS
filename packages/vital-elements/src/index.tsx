/**
 * Copyright Â© 2022 Johnson & Johnson
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * http://www.apache.org/licenses/LICENSE-2.0
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export * from './components/AccordionElement';
export * from './components/ArticleCardElement';
export * from './components/BreadcrumbElement';
export * from './components/ButtonElement';
export * from './components/CategoryCardElement';
export * from './components/Color';
export * from './components/DividerElement';
export * from './components/FilterElement';
export * from './components/FloatingButtonElement';
export * from './components/FontSize';
export * from './components/FooterElement';
export * from './components/HeroBannerElement';
export * from './components/ImageCarouselElement';
export * from './components/InputElement';
export * from './components/KnapsackDemoElement';
export * from './components/LinkElement';
export * from './components/ModalElement';
export * from './components/NavigationElement';
export * from './components/PDPProductOverviewElement';
export * from './components/ProductCardElement';
export * from './components/ScrollIndicatorElement';
export * from './components/Spacing';
export * from './components/TextDecoration';
export * from './components/KnapsackDemoElement';
export { default as vitalScrollIndicatorElement } from './generated/vitalScrollIndicatorElement';
export * from './components/Typography';
export * from './util';
