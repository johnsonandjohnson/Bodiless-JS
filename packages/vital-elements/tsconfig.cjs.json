{
  "extends": "../../tsconfig.settings.json",
  "compilerOptions": {
    "module": "commonjs",
    "rootDir": "src",
    "outDir": "./lib/cjs",
    "baseUrl": "."
  },
  "include": [
    "src/util",
  ],
  "exclude": [
    "lib",
    "node_modules",
    "__tests__/**/*.*",
    "src/**/__tests__/**/*.*"
  ]
}
