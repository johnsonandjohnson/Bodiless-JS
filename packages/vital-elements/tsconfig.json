{
  "extends": "../../tsconfig.settings.json",
  "compilerOptions": {
    "rootDir": "src",
    "outDir": "lib",
    "baseUrl": "."
  },
  "include": [
    "src",
    "../../@types/*",
  ],
  "exclude": [
    "lib",
    "node_modules",
    "__tests__/**/*.*",
    "src/**/__tests__/**/*.*"
  ]
}