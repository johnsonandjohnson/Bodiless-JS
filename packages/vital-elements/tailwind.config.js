/* eslint-disable import/no-dynamic-require, global-require */
/**
 * Copyright Â© 2021 Johnson & Johnson
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * http://www.apache.org/licenses/LICENSE-2.0
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { getPackageTailwindConfig } from '@bodiless/fclasses';

const tailwindcssDir = require('tailwindcss-dir')();
const designTokens = require('./assets/design-tokens.nested.json');

// Parse the JSON Tokens data
const parsedDesignTokens = JSON.parse(JSON.stringify(designTokens));

/**
 * The `design-tokens.nested.json` file should only contain all Core tokens.
 * These core tokens are then used to extend the corresponding Tailwind config sections.
 *
 * @TODO: The `border.style` tokens do not have a corresponding key in the Tailwind config.
 * `borderStyle` is only a valid key for the TW variants and plugins. The default TW values for
 * borderStyle match the provided tokens.
 *
 * @TODO: `typography` tokens include `text-decoration` and `text-transform` tokens which are
 * not intended to be extended or modified in the Tailwind config and do not have corresponding
 * keys in the Tailwind config. The default TW config covers these tokens and provides the same
 * Default classes like `capitalize` or `uppercase`.
 *
 * @TODO: Also note that currently the `design-tokens.nested.json` is manually adjusted to
 * the more favourable format to resolve certain issues with resolving Figma Plugin aliases.
 * Thehere is a script being worked on to format tokens automatically.
 *
 * @TODO: The `border-width` and `border-style` tokens in JSON are combined under the same key.
 * Tailwind treats it as two separate styles with different config key. In manually updated JSON
 * it is split into two different token "section" like `border.width` and `boder.style`
 * to avoid TW config issues.
 */
const {
  // colors,
  // spacing,
  // typography,
  opacity,
  // border,
} = parsedDesignTokens;

const resolver = (pkgName) => require.resolve(pkgName);

const twConfig = {
  content: [
    './lib/**/!(*.d).{ts,js,jsx,tsx}',
  ],
  theme: {
    fontSize: {
      xs: ['0.75rem', '1.125rem'],
      sm: ['0.875rem', '1.25rem'],
      base: ['1rem', '1.375rem'],
      lg: ['1.75rem', '1.875rem'],
      xl: ['1.625rem', '2rem'],
      '2xl': ['2rem', '2.25rem'],
      '3xl': ['2.563rem', '2.813rem'],
      'm-xs': ['0.688rem', '0.938rem'],
      'm-sm': ['0.75rem', '1rem'],
      'm-base': ['0.875rem', '1.125rem'],
      'm-lg': ['1.125rem', '1.375rem'],
      'm-xl': ['1.438rem', '1.625rem'],
      'm-2xl': ['1.813rem', '2.125rem'],
      'm-3xl': ['2.25rem', '2.5rem'],
    },
    colors: {
      transparent: 'transparent',
      'vital-primary': {
        brand: '#CA081B',
        'card-bg': '#ffffff',
        'page-bg': '#F4F4F4',
        interactive: '#019881',
        'interactive-active': '#000341',
        divider: '#D8D8D8',
        'body-copy': '#63666A',
        'header-copy': '#212121',
        'footer-copy': '#FFFFFF',
        'button-selected': '#015B4D',
        'button-disabled': '#999999',
      },
      'vital-secondary': {
        eyebrow: '#CC0099',
        'footer-bg': '#2B2B33',
        interactive: '#018571',
        table: '#E5E7EB',
        'table-interactive': '#f3f4f6',
        separator: '#666666',
        search: '#888888',
        'button-selected': '#B5E1DA',
        'button-hovered': '#E1F3F0',
        'button-disabled': '#999999',
        'border-button-selected': '#015B4D',
      },
      white: '#FFFFFF',
    },
    extend: {
      /**
       * Vital 2.0 Tokens coming from Figma plugin.
       */
      colors: {
        // ...colors,
        ColorButtonPrimaryTextLightThemeText: '#ffffff',
        ColorInteractiveLightThemeIdle: '#019881',
        ColorInteractiveLightThemeHover: '#017A67',
        ColorInteractiveLightThemeFocus: '#017A67',
        ColorInteractiveLightThemePressed: '#015B4D',
        ColorInteractiveLightThemeDisabled: '#999999',
      },
      fontFamily: {
        DMSans: ['DM Sans', 'sans-serif'],
        // ...typography['font-family'],
        1: 'DM Sans',
        2: 'DM Sans',
      },
      fontWeights: {
        // ...typography.font,
      },
      fontSize: {
        '14px': '14px',
        '16px': '16px',
      },
      lineHeight: {
        // ...typography.leading,
        5: '120%',
        6: '150%'
      },
      textSizes: {
        // ...typography.text,
      },
      spacing: {
        // ...spacing,
        '0px': '0px',
        '1px': '1px',
        '2px': '2px',
        '3px': '3px',
        '4px': '4px',
        '5px': '4px',
        '6px': '4px',
        '7px': '4px',
        '8px': '4px',
        '9px': '4px',
        '10px': '4px',
        '12px': '4px',
        '14px': '4px',
        '16px': '4px',
        '18px': '4px',
        '20px': '4px',
        '24px': '4px',
        '32px': '4px',
        '36px': '4px',
        '40px': '4px',
        '48px': '4px',
        '64px': '4px',
      },
      borderWidth: {
        // ...border.width,
        '1px': '1px',
      },
      borderRadius: {
        // ...border.rounded,
        0: '0px',
        2: '2px',
        4: '4px',
        6: '6px',
        8: '8px',
        10: '10px',
        12: '12px',
        20: '20px',
        150: '150px',
        600: '600px',
        pill: '9999px',
      },
      opacity: {
        // ...opacity,
      },
    },
  },
  plugins: [
    tailwindcssDir,
  ],
};

module.exports = getPackageTailwindConfig({
  twConfig,
  resolver,
});
