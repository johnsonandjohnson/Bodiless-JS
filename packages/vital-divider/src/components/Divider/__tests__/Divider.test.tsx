import React from 'react';
import { testTokens } from '@bodiless/vital-elements';
import { DividerClean, vitalDividers} from '..';

// eslint-disable-next-line jest/valid-describe
describe('Divider Tokens', testTokens(DividerClean, vitalDividers));
