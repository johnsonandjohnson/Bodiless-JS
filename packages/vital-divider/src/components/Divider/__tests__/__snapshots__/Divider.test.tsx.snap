// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Tokens Matches snapshot for Base 1`] = `
<div>
  <hr
    className="border-1px border-solid"
  />
</div>
`;

exports[`Divider Tokens Matches snapshot for WithPrimaryDivider 1`] = `
<div>
  <hr
    className="border-kenvue-neutrals-light-grey"
  />
</div>
`;

exports[`Divider Tokens Matches snapshot for WithSecondaryDivider 1`] = `
<div>
  <hr
    className="border-kenvue-neutrals-grey"
  />
</div>
`;

exports[`Divider Tokens Matches snapshot for WithVerticalOrientation 1`] = `
<div>
  <hr
    className="rotate-90 h-0px"
  />
</div>
`;
